pkg/assembler/helpers/purl_test.go

```diff
@@ -207,6 +207,328 @@ func TestPurlConvert(t *testing.T) {
+			version:         "1.2.3",
+			subpath:         "",
+			qualifiers:      []string{"repository_url", "https://myrepo.example.com"},
+		}, {
+			expectedPurlUri: "pkg:hex/jason@1.1.2",
+			pkgType:         "hex",
+			namespace:       "",
+			name:            "jason",
+			version:         "1.1.2",
+			subpath:         "",
+			qualifiers:      []string{},
+		}, {
+			expectedPurlUri: "pkg:huggingface/distilbert-base-uncased@043235d6088ecd3dd5fb5ca3592b6913fd516027",
+			pkgType:         "huggingface",
+			namespace:       "",
+			name:            "distilbert-base-uncased",
+			version:         "043235d6088ecd3dd5fb5ca3592b6913fd516027",
+			subpath:         "",
+			qualifiers:      []string{},
+		}, {
+			expectedPurlUri: "pkg:maven/org.apache.xmlgraphics/batik-anim@1.9.1?type=zip&classifier=dist",
+			pkgType:         "maven",
+			namespace:       "org.apache.xmlgraphics",
+			name:            "batik-anim",
+			version:         "1.9.1",
+			subpath:         "",
+			qualifiers:      []string{"type", "zip", "classifier", "dist"},
+		}, {
+			expectedPurlUri: "pkg:mlflow/trafficsigns@10?model_uuid=36233173b22f4c89b451f1228d700d49&run_id=410a3121-2709-4f88-98dd-dba0ef056b0a&repository_url=https:%2F%2Fadb-5245952564735461.0.azuredatabricks.net%2Fapi%2F2.0%2Fmlflow",
+			pkgType:         "mlflow",
+			namespace:       "",
+			name:            "trafficsigns",
+			version:         "10",
+			subpath:         "",
+			qualifiers:      []string{"model_uuid", "36233173b22f4c89b451f1228d700d49", "run_id", "410a3121-2709-4f88-98dd-dba0ef056b0a", "repository_url", "https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow"},
+		}, {
+			expectedPurlUri: "pkg:npm/foobar@12.3.1",
+			pkgType:         "npm",
+			namespace:       "",
+			name:            "foobar",
+			version:         "12.3.1",
+			subpath:         "",
+			qualifiers:      []string{},
+		}, {
+			expectedPurlUri: "pkg:npm/%40angular/animation@12.3.1",
+			pkgType:         "npm",
+			namespace:       "@angular",
+			name:            "animation",
+			version:         "12.3.1",
+			subpath:         "",
+			qualifiers:      []string{},
+		}, {
+			expectedPurlUri: "pkg:nuget/EnterpriseLibrary.Common@6.0.1304",
+			pkgType:         "nuget",
+			namespace:       "",
+			name:            "EnterpriseLibrary.Common",
+			version:         "6.0.1304",
+			subpath:         "",
+			qualifiers:      []string{},
+		}, {
+			expectedPurlUri: "pkg:qpkg/blackberry/com.qnx.sdp@7.0.0.SGA201702151847",
+			pkgType:         "qpkg",
+			namespace:       "blackberry",
+			name:            "com.qnx.sdp",
+			version:         "7.0.0.SGA201702151847",
+			subpath:         "",
+			qualifiers:      []string{},
+		}, {
+			// Special OCI case
+			//TODO: similar issue to above.
+
+			//expectedPurlUri: "pkg:oci/debian@sha256%3A244fd47e07d10?repository_url=docker.io/library/debian&arch=amd64&tag=latest",
+			expectedPurlUri: "pkg:oci/docker.io/library/debian@sha256:244fd47e07d10?arch=amd64&tag=latest",
```

Comment:
this one is actually pretty tricky, technically the PURL has the repository in the qualifiers (it’s kind of weird, but as long as OSV can recognize it i think its fine we don’t particularly need a 1:1 mapping

Reply:
Yeah not sure how we want to handle this issue. Seems like the library does it differently from the examples. So which way is right?

Reply:
I will link the issue.

Reply:
Opened issue #635